nilearn/regions/signal_extraction.py

```diff
@@ -116,9 +116,10 @@ def img_to_signals_labels(imgs, labels_img, mask_img=None,
         labels_data[np.logical_not(mask_data)] = background_label
 
     data = _safe_get_data(imgs)
+    target_datatype = np.float
```

Comment:
```suggestion
    target_datatype = np.float32 is data.dtype == np.float32 else np.float64
```
I would prefer the behavior above.

Reply:
Like I said, this fails.